TombEngine/Game/control/box.cpp

```diff
@@ -137,77 +138,68 @@ bool MoveCreature3DPos(Pose* fromPose, Pose* toPose, int velocity, short angleDi
 	return false;
 }
 
-void CreatureYRot2(Pose* fromPose, short angle, short angleAdd) 
+void CreatureYRot2(Pose* fromPose, short angle, short angleAdd)
 {
-	if (angleAdd < angle)
+	if (angle > angleAdd)
 	{
 		fromPose->Orientation.y += angleAdd;
-		return;
-	} 
-
-	if (angle < -angleAdd)
+	}
+	else if (angle < -angleAdd)
 	{
 		fromPose->Orientation.y -= angleAdd;
-		return;
-	} 
-
-	fromPose->Orientation.y += angle;
+	}
+	else
+	{
+		fromPose->Orientation.y += angle;
+	}
 }
 
 bool SameZone(CreatureInfo* creature, ItemInfo* target)
 {
-	int* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
-	auto* item = &g_Level.Items[creature->ItemNumber];
-
-	auto* room = &g_Level.Rooms[item->RoomNumber];
-	FloorInfo* floor = GetSector(room, item->Pose.Position.x - room->x, item->Pose.Position.z - room->z);
-	if (floor->Box == NO_BOX)
-		return false;
-
-	item->BoxNumber = floor->Box;
+	auto& item = g_Level.Items[creature->ItemNumber];
+	auto* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
 
-	room = &g_Level.Rooms[target->RoomNumber];
-	floor = GetSector(room, target->Pose.Position.x - room->x, target->Pose.Position.z - room->z);
-
-	if (floor->Box == NO_BOX)
-		return false;
+	auto& room1 = g_Level.Rooms[item.RoomNumber];
+	item.BoxNumber = GetSector(&room1, item.Pose.Position.x - room1.z, item.Pose.Position.z - room1.z)->Box;
 
-	target->BoxNumber = floor->Box;
+	auto& room2 = g_Level.Rooms[target->RoomNumber];
+	target->BoxNumber = GetSector(&room2, target->Pose.Position.x - room2.z, target->Pose.Position.z - room2.z)->Box;
 
-	return (zone[item->BoxNumber] == zone[target->BoxNumber]);
+	return (zone[item.BoxNumber] == zone[target->BoxNumber]);
 }
 
 short AIGuard(CreatureInfo* creature) 
 {
-	auto* item = &g_Level.Items[creature->ItemNumber];
-	if (item->AIBits & (GUARD | PATROL1))
+	auto& item = g_Level.Items[creature->ItemNumber];
+	if (item.AIBits & MODIFY)
 		return 0;
 
-	int random = GetRandomControl();
-
-	if (random < 256)
+	if (Random::TestProbability(1 / 4.0f))
```

Comment:
Are you sure probabilities are right? `GetRandomControl` returns values on a range from 0 to 0xFFFF. So `384` here is not equal to `1 / 3.0f` in my opinion (and down below).